packages/components/src/internal/components/lineage/models.ts

```diff
@@ -791,11 +792,28 @@ function createVisNode(
     // show the alternate icon image color if this node is the seed or has been selected
     const { image, imageBackup, imageSelected, imageShape } = node.iconProps;
 
+    let nodeLabel = node.name;
+    // Issue 51432: LKSM: special character in sample names result in client side exception: SyntaxError: unterminated character class
+    // vis-network does special processing for labels that evaluates to true for `/&/.test()`
+    if (nodeLabel?.indexOf('&') > -1) {
+        // for labels that starts with '&', the entire label is replaced with '&lt;' by vis
+        if (nodeLabel.startsWith('&')) {
+            nodeLabel = nodeLabel.replace('&', FULLWIDTH_AMPERSAND);
+        }
+
+        try {
+            new RegExp(nodeLabel);
```

Comment:
Probably not a huge deal, but since regular expressions and exception throwing are relatively expensive, perhaps it would be more efficient to always do the `replaceAll`. Or are there cases where this substitution isn't a good thing?

Reply:
I was hoping to preserve the original labels as much as possible. The hope is that there aren't many that starts with & to trigger this code.

Reply:
Probably true that there aren't many &-names, or we would have heard more about this problem, and maybe the world of JavaScript is different in this regard than Java, but I still think it's more efficient (and targeted) to do the replacement without regex construction, especially since it seems like this RegExp construction could throw an error for other reasons than the &.

Reply:
Sure that makes sense. Since user can't copy label from the vis canvas, it's no harm to do the replacement more aggressively. Updated to use replaceAll instead.